components/calendar/src/calendar-input/calendar-input.js

```diff
@@ -76,14 +68,36 @@ export const CalendarInput = ({
 
     return (
         <>
-            <div ref={ref}>
+            <div className="calendar-input-wrapper" ref={ref}>
                 <InputField
                     label="Pick a date"
                     {...rest}
                     type="text"
                     onFocus={onFocus}
-                    value={value}
+                    value={date}
                 />
+                {clearable && (
+                    <div
+                        className={cx('calendar-clear-button', {
+                            // ToDo: this is a workaround to show the clear button in the correct place when an icon is shown.
+                            // Long-term, we should abstract and share the logic multi-select uses for the input-wrapper
+                            // https://dhis2.atlassian.net/browse/DHIS2-14848
+                            'with-icon':
+                                rest.valid || rest.error || rest.warning,
```

Comment:
I don't see any practical use cases, but just for completeness sake I think it'd be good to add `rest.loading`, since a loading input will also be suffixed with an icon, see here: https://github.com/dhis2/ui/blob/d0c497911733c8185126dd2c43d6a2069040c342/components/input/src/input/input.js#L182-L187